pkg/collector/scheduler/scheduler.go

```diff
@@ -110,6 +110,7 @@ func (s *Scheduler) Cancel(id check.ID) error {
 	if err != nil {
 		return fmt.Errorf("unable to remove the Job from the queue: %s", err)
 	}
+	delete(s.checkToQueue, id)
```

Comment:
nice catch 👌 

Originally an oversight from https://github.com/DataDog/datadog-agent/pull/2127, but at that point only caused a very small mem leak in very dynamic AD environments. With the new logic in this PR, this oversight caused a new problem now that we use this map to determine if a check is still in the schedule (in `IsCheckScheduled`)